src/connection.rs

```diff
@@ -152,7 +154,10 @@ async fn requests(topic: String, payload_size: usize, count: usize, requests_tx:
 
         let publish = PublishRaw::new(&topic, qos, payload).unwrap();
         let publish = Request::PublishRaw(publish);
-        requests_tx.send(publish).await.unwrap();
+        interval.tick().await;
+        if let Err(_) = requests_tx.send(publish).await {
```

Comment:
It's probably better to unwrap. Error here is usually a bug.

Reply:
Then we are seeing too much of this works. After just a bunch of push, this starts throwing errr.

Reply:
Ok. I'll merge this and see what's happening